errs/error.go

```diff
@@ -0,0 +1,249 @@
+package errs
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// StatusCoder interface is used by errors that returns the HTTP response code.
+type StatusCoder interface {
+	StatusCode() int
+}
+
+// StackTracer must be by those errors that return an stack trace.
+type StackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+// Option modifies the Error type.
+type Option func(e *Error) error
+
+// WithMessage returns an Option that modifies the error by overwriting the
+// message only if it is empty.
+func WithMessage(format string, args ...interface{}) Option {
+	return func(e *Error) error {
+		if len(e.Msg) > 0 {
+			return e
+		}
+		e.Msg = fmt.Sprintf(format, args...)
+		return e
+	}
+}
+
+// Error represents the CA API errors.
+type Error struct {
+	Status int
+	Err    error
+	Msg    string
+}
+
+// New returns a new Error. If the given error implements the StatusCoder
+// interface we will ignore the given status.
+func New(status int, err error, opts ...Option) error {
+	var e *Error
+	if sc, ok := err.(StatusCoder); ok {
+		e = &Error{Status: sc.StatusCode(), Err: err}
+	} else {
+		cause := errors.Cause(err)
+		if sc, ok := cause.(StatusCoder); ok {
+			e = &Error{Status: sc.StatusCode(), Err: err}
+		} else {
+			e = &Error{Status: status, Err: err}
+		}
+	}
+	for _, o := range opts {
+		o(e)
+	}
+	return e
+}
+
+// ErrorResponse represents an error in JSON format.
+type ErrorResponse struct {
+	Status  int    `json:"status"`
+	Message string `json:"message"`
+}
+
+// Cause implements the errors.Causer interface and returns the original error.
+func (e *Error) Cause() error {
+	return e.Err
+}
```

Comment:
Not required right now, but from Go 1.13, the new way to do it is using an `Unwrap() error`, this is not supported by `pkg/errors`, we could just have both `Cause` and `Unwrap`.